### PR TITLE
Sync initial bookmarks in order

### PR DESCRIPTION
**Pull request by ayumi**
Re-introduce SyncRecord.Bookmark.index after Sites Map refactor.

Fix https://github.com/brave/sync/issues/45

Auditors: @diracdeltas

Test plan:
1. Start Pyramid 0. It should have bookmarks and bookmark folders.
2. Enable Sync on Pyramid 0, and restart to complete setup.
3. Start Pyramid 1. Add it to Pyramid 0's Sync profile and restart.
4. Bookmarks should sync onto Pyramid 1 in the same order as Pyd 0.

### CODE REVIEW

File: app/sync.js [hunk 1]
```diff
@@ -165,38 +165,48 @@ module.exports.onSyncReady = (isFirstRun, e) => {
...
-   * folder if needed.
+   * folder if needed. For folders, set and memoize folderId to ensure
+   * consistent parentFolderObjectIds.
+   * Otherwise siteUtil.createSiteData() will generate new objectIds every
+   * call; there's not enough time to dispatch id updates to appStore.sites.
    * @param {Immutable.Map} site
+   * @param {number} index Sets the index for SyncRecord.Bookmark.index
+   *   so other clients apply records in a batch in the correct order.
+   * @returns {number} Next index to use. Note this may not be index +1
+   *   because for foldered bookmarks we first recurse and sync parent folders.
```
Comment (diracdeltas) [inline app/sync.js]: looks like this jsdoc is no longer correct?

Comment (ayumi) [inline app/sync.js]: good 🔼

**Comment by diracdeltas** [comment]
i noticed that not all of my bookmarks were synced in the correct folders :(. but this is also happening on feature/syncing so not related to this PR.
top: pyramid 0
bottom: pyramid 1
screen shot 2017-02-01 at 1 28 58 am

**Comment by luixxiul** [comment]
I added the label 0.13.3 in case that the sync would be implemented with that version. Let's reset it later if it is not.

**Pull request merged**